### PR TITLE
[mssql] Escape all columns that do not start with "__artie"

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/sql/escape.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func NeedsEscaping(name string, destKind constants.DestinationKind) bool {
 	if destKind == constants.Redshift {
 		reservedKeywords = constants.RedshiftReservedKeywords
 	} else if destKind == constants.MSSQL {
-		reservedKeywords = constants.MSSQLReservedKeywords
+		return !strings.HasPrefix(name, constants.ArtiePrefix)
```
Comment (Tang8330) [inline lib/sql/escape.go:28]: Why not just escape all?

**Review by Tang8330 — APPROVED** [review]

File: lib/sql/escape.go (hunk 1 above)
Comment (nathan-artie) [reply lib/sql/escape.go:28]: If we escape all we get test that say `artie delete flag doesn't exist`.

Comment (nathan-artie) [reply lib/sql/escape.go:28]: I'm guessing there must be some places where we escape the column names and then check if the column names contain a specific value.

Comment (Tang8330) [reply lib/sql/escape.go:28]: That's a guardrail in `dml/merge`

Comment (Tang8330) [reply lib/sql/escape.go:28]: The comparison is running `"__artie_deleted` == __artie_deleted` which will always equate to false

**Pull request merged**